### PR TITLE
Adyen are stopping support for SHA1 HPP pages.

### PR DESCRIPTION
**Pull request by SebAshton**
On July 1st Adyen will no longer support SHA1 in favour of SHA256 encryption. The community behind the Adyen gem are working towards a fix for this in their `HPP` branch https://github.com/wvanbergen/adyen/issues/118

This WIP PR removes the Adyen version lock, when testing if the adyen gem doesn't upgrade automatically the following to your `Gemfile` beneath where you declare `solidus-adyen` e.g.

```
gem 'solidus-adyen', github: 'madetech/solidus-adyen', branch: 'upgrade-adyen-version'
gem 'adyen', github: 'wvanbergen/adyen', branch: 'hpp'
```

Will integrate and report back.

### CODE REVIEW

**Comment by mvz** [comment]
You will need to use Adyen::HPP instead of Adyen::Form as well.

**Comment by SebAshton** [comment]
Thank you for the pointer @mvz

File: lib/spree/adyen/form.rb [hunk 1]
```diff
@@ -132,7 +132,8 @@ def order_params order
             merchant_reference: order.number.to_s,
             country_code: order.billing_address.country.iso,
             payment_amount: (order.total * 100).to_int,
-            shopper_locale: I18n.locale.to_s.gsub("-", "_")
+            shopper_locale: I18n.locale.to_s.gsub("-", "_"),
+            shopper_email: order.email
```
Comment (Dkendal) [inline lib/spree/adyen/form.rb:135]: is this line intended to be added to this pr?

Comment (SebAshton) [inline lib/spree/adyen/form.rb:135]: Yes, the shopper_email is now required to calculate a valid Merchant Signature. Please hold fire on merging as this PR is still not ready to merge.

File: solidus-adyen.gemspec [hunk 2]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_runtime_dependency "adyen", "~> 1.4"
+  spec.add_runtime_dependency "adyen"
```
Comment (Dkendal) [inline solidus-adyen.gemspec]: Shouldn't this be capped at 2.0?

File: lib/spree/adyen/form.rb (hunk 1 above)
Comment (Dkendal) [inline lib/spree/adyen/form.rb:135]: ok

**Comment by SebAshton** [comment]
The team behind https://github.com/wvanbergen/adyen are looking to release the HPP updates this morning. When this has been done I'll update the gem dependencies, and we'll be good to merge.

File: lib/spree/adyen/form.rb (hunk 1 above)
Comment (luukveenis) [inline lib/spree/adyen/form.rb:135]: The shopper email is actually only a required field for recurring payments:
https://docs.adyen.com/developers/hpp-manual#hpppaymentfields
With the SHA1 signing method it needs to be included when calculating the signature only if it is present.

Comment (SebAshton) [inline lib/spree/adyen/form.rb:135]: While I was testing this the HMAC Calculation didn't match without it. Below is a screenshot from the "Test HMAC" tab from the skin page. It shows it as a required field.

screen shot 2016-06-18 at 11 52 15

Also, the shopper email is also used for velocity fraud checks, so I don't see any harm in having it included either way. More fraud protection is surely a good thing?

**Comment by chrisradford** [comment]
@SebAshton any updates on this? Quite aware the initial PR on the https://github.com/wvanbergen/adyen repo mentioned July 1st was the cut-off date for switching this over.

**Comment by SebAshton** [comment]
@chrisradford I was doing the necessary work this morning, and got distracted. Committing in now.

**Comment by SebAshton** [comment]
@Dkendal this is good from my side. FWIW we've been using this branch in production now for well over a week without issue.

**Comment by jordan-brough** [comment]
@Dkendal are you 👍 on this?

**Comment by Dkendal** [comment]
Alright this looks a-okay to me :+1:

**Comment by jordan-brough** [comment]
@Dkendal thanks!

**Comment by Dkendal** [comment]
i have just cut a new major release (1.0.0)

**Comment by SebAshton** [comment]
Does this close #34?

**Comment by Dkendal** [comment]
yes, thank you @SebAshton

**Pull request merged**